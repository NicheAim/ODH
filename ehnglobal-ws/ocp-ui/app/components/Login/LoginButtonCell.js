import { Cell } from 'styled-css-grid';

const LoginButtonCell = Cell.extend`
  padding: 25px 0;
`;

LoginButtonCell.propTypes = {};

export default LoginButtonCell;
