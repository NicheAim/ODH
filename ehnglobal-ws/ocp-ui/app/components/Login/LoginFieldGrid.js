import { Grid } from 'styled-css-grid';

const LoginFieldGrid = Grid.extend`
  margin: 0 50px;
`;

LoginFieldGrid.propTypes = {};

export default LoginFieldGrid;
