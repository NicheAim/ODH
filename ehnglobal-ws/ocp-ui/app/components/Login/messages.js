/*
 * Login Messages
 *
 * This contains all the text for the Login component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'ocpui.components.Login.title',
    defaultMessage: 'Login to your account',
  },
  viewUserAccounts: {
    id: 'ocpui.components.Login.viewUserAccounts',
    defaultMessage: 'Sample User Login Details',
  },
  forgotLink: {
    id: 'ocpui.components.Login.forgotLink',
    defaultMessage: 'Forgot your username or password?',
  },
  validation: {
    required: {
      id: 'ocpui.components.Login.loginForm.validation.required',
      defaultMessage: 'Required',
    },
  },
  loginButton: {
    id: 'ocpui.components.Login.loginButton',
    defaultMessage: 'Login',
  },
  authenticatingButton: {
    id: 'ocpui.components.Login.authenticatingButton',
    defaultMessage: 'Login...',
  },
  hintText: {
    username: {
      id: 'ocpui.components.Login.loginForm.hintText.username',
      defaultMessage: 'Username',
    },
    password: {
      id: 'ocpui.components.Login.loginForm.hintText.password',
      defaultMessage: 'Password',
    },
  },
  floatingLabelText: {
    username: {
      id: 'ocpui.components.Login.loginForm.floatingLabelText.username',
      defaultMessage: 'Username',
    },
    password: {
      id: 'ocpui.components.Login.loginForm.floatingLabelText.password',
      defaultMessage: 'Password',
    },
  },
});
