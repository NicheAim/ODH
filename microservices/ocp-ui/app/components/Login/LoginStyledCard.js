/**
 *
 * LoginStyledCard
 *
 */
import styled from 'styled-components';
import { Card } from 'material-ui/Card';

const LoginStyledCard = styled(Card).attrs({
  style: {
    backgroundColor: 'rgba(242, 242, 242, 1)',
    borderWidth: '1px',
  },
})('');

LoginStyledCard.propTypes = {};

export default LoginStyledCard;
