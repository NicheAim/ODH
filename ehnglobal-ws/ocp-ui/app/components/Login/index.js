/**
 *
 * Login
 *
 */

import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Form, Formik } from 'formik';
import yup from 'yup';
import { FormattedMessage } from 'react-intl';
import { Cell, Grid } from 'styled-css-grid';
import CardHeader from 'components/CardHeader';
import TextField from 'components/TextField';
import StyledRaisedButton from 'components/StyledRaisedButton';
import LoginStyledCard from './LoginStyledCard';
import LoginFieldGrid from './LoginFieldGrid';
import LoginButtonCell from './LoginButtonCell';
import messages from './messages';

function Login(props) {
  const { onLogin, isShowSampleUserLoginDetails } = props;

  return (
    <Grid columns={3}>
      <Cell left={2} top={3} center>
        <LoginStyledCard>
          <CardHeader title={<FormattedMessage {...messages.title} />} />
          <Formik
            onSubmit={(values, actions) => {
              onLogin(values, actions);
            }}
            validationSchema={yup.object().shape({
              username: yup.string()
                .required((<FormattedMessage {...messages.validation.required} />)),
              password: yup.string()
                .required((<FormattedMessage {...messages.validation.required} />)),
            })}
            render={(loginFormProps) => {
              const { isSubmitting, dirty, isValid } = loginFormProps;
              return (
                <Form>
                  <LoginFieldGrid
                    columns={1}
                    rows="120px 120px 45px 100px"
                    areas={[
                      'username',
                      'password',
                      'forgotLink',
                      'loginButton',
                    ]}
                  >
                    <Cell>
                      <TextField
                        name="username"
                        hintText={<FormattedMessage {...messages.hintText.username} />}
                        floatingLabelText={<FormattedMessage {...messages.floatingLabelText.username} />}
                        fullWidth
                        floatingLabelFixed
                      />
                    </Cell>
                    <Cell>
                      <TextField
                        name="password"
                        type="password"
                        hintText={<FormattedMessage {...messages.hintText.password} />}
                        floatingLabelText={<FormattedMessage {...messages.floatingLabelText.password} />}
                        fullWidth
                        floatingLabelFixed
                      />
                    </Cell>
                    <Cell>
                      {isShowSampleUserLoginDetails ?
                        <Link to="/ocp-ui/sample-user-login-details" target="_blank">
                          <FormattedMessage {...messages.viewUserAccounts} />
                        </Link> : ''
                      }
                    </Cell>
                    <LoginButtonCell>
                      <StyledRaisedButton
                        type="submit"
                        fullWidth
                        disabled={!dirty || isSubmitting || !isValid}
                      >{isSubmitting ?
                        <FormattedMessage {...messages.authenticatingButton} /> :
                        <FormattedMessage {...messages.loginButton} />}
                      </StyledRaisedButton>
                    </LoginButtonCell>
                  </LoginFieldGrid>
                </Form>
              );
            }}
          />
        </LoginStyledCard>
      </Cell>
    </Grid>
  );
}

Login.propTypes = {
  onLogin: PropTypes.func.isRequired,
  isShowSampleUserLoginDetails: PropTypes.bool,
};

export default Login;
