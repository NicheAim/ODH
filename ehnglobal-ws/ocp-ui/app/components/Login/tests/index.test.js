import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import Login from '../index';

configure({ adapter: new Adapter() });

describe('<Login />', () => {
  it('should match snapshot', () => {
    // Arrange
    const onLogin = jest.fn();

    // Act
    const renderedComponent = shallow(<Login onLogin={onLogin} />);

    // Assert
    expect(renderedComponent).toMatchSnapshot();
  });
});
