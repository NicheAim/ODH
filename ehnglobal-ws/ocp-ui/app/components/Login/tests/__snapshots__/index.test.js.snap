// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    onLogin={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Cell__Cell
        center={true}
        left={2}
        top={3}
      >
        <Styled(Card)>
          <CardHeader
            title={
              <FormattedMessage
                defaultMessage="Login to your account"
                id="ocpui.components.Login.title"
                values={Object {}}
              />
            }
          />
          <Formik
            enableReinitialize={false}
            isInitialValid={false}
            onSubmit={[Function]}
            render={[Function]}
            validateOnBlur={true}
            validateOnChange={true}
            validationSchema={
              ObjectSchema {
                "_blacklist": Array [],
                "_conditions": Array [],
                "_defaultDefault": [Function],
                "_deps": Array [],
                "_excludedEdges": Array [],
                "_exclusive": Object {},
                "_mutate": false,
                "_nodes": Array [
                  "password",
                  "username",
                ],
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "object",
                "_typeError": [Function],
                "_whitelist": Array [],
                "fields": Object {
                  "password": StringSchema {
                    "_blacklist": Array [],
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": Array [],
                    "tests": Array [
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                  "username": StringSchema {
                    "_blacklist": Array [],
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": Array [],
                    "tests": Array [
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                },
                "tests": Array [],
                "transforms": Array [
                  [Function],
                ],
              }
            }
          />
        </Styled(Card)>
      </Cell__Cell>,
      "columns": 3,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "center": true,
        "children": <Styled(Card)>
          <CardHeader
            title={
              <FormattedMessage
                defaultMessage="Login to your account"
                id="ocpui.components.Login.title"
                values={Object {}}
              />
            }
          />
          <Formik
            enableReinitialize={false}
            isInitialValid={false}
            onSubmit={[Function]}
            render={[Function]}
            validateOnBlur={true}
            validateOnChange={true}
            validationSchema={
              ObjectSchema {
                "_blacklist": Array [],
                "_conditions": Array [],
                "_defaultDefault": [Function],
                "_deps": Array [],
                "_excludedEdges": Array [],
                "_exclusive": Object {},
                "_mutate": false,
                "_nodes": Array [
                  "password",
                  "username",
                ],
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "object",
                "_typeError": [Function],
                "_whitelist": Array [],
                "fields": Object {
                  "password": StringSchema {
                    "_blacklist": Array [],
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": Array [],
                    "tests": Array [
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                  "username": StringSchema {
                    "_blacklist": Array [],
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": Array [],
                    "tests": Array [
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                },
                "tests": Array [],
                "transforms": Array [
                  [Function],
                ],
              }
            }
          />
        </Styled(Card)>,
        "left": 2,
        "top": 3,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <CardHeader
              title={
                <FormattedMessage
                  defaultMessage="Login to your account"
                  id="ocpui.components.Login.title"
                  values={Object {}}
                />
              }
            />,
            <Formik
              enableReinitialize={false}
              isInitialValid={false}
              onSubmit={[Function]}
              render={[Function]}
              validateOnBlur={true}
              validateOnChange={true}
              validationSchema={
                ObjectSchema {
                  "_blacklist": Array [],
                  "_conditions": Array [],
                  "_defaultDefault": [Function],
                  "_deps": Array [],
                  "_excludedEdges": Array [],
                  "_exclusive": Object {},
                  "_mutate": false,
                  "_nodes": Array [
                    "password",
                    "username",
                  ],
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "object",
                  "_typeError": [Function],
                  "_whitelist": Array [],
                  "fields": Object {
                    "password": StringSchema {
                      "_blacklist": Array [],
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "required": false,
                      },
                      "_mutate": false,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "string",
                      "_typeError": [Function],
                      "_whitelist": Array [],
                      "tests": Array [
                        [Function],
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                    "username": StringSchema {
                      "_blacklist": Array [],
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "required": false,
                      },
                      "_mutate": false,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "string",
                      "_typeError": [Function],
                      "_whitelist": Array [],
                      "tests": Array [
                        [Function],
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                  },
                  "tests": Array [],
                  "transforms": Array [
                    [Function],
                  ],
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": <FormattedMessage
                defaultMessage="Login to your account"
                id="ocpui.components.Login.title"
                values={Object {}}
              />,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "enableReinitialize": false,
              "isInitialValid": false,
              "onSubmit": [Function],
              "render": [Function],
              "validateOnBlur": true,
              "validateOnChange": true,
              "validationSchema": ObjectSchema {
                "_blacklist": Array [],
                "_conditions": Array [],
                "_defaultDefault": [Function],
                "_deps": Array [],
                "_excludedEdges": Array [],
                "_exclusive": Object {},
                "_mutate": false,
                "_nodes": Array [
                  "password",
                  "username",
                ],
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "object",
                "_typeError": [Function],
                "_whitelist": Array [],
                "fields": Object {
                  "password": StringSchema {
                    "_blacklist": Array [],
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": Array [],
                    "tests": Array [
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                  "username": StringSchema {
                    "_blacklist": Array [],
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": false,
                    },
                    "_mutate": false,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": Array [],
                    "tests": Array [
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                  },
                },
                "tests": Array [],
                "transforms": Array [
                  [Function],
                ],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Cell__Cell
          center={true}
          left={2}
          top={3}
        >
          <Styled(Card)>
            <CardHeader
              title={
                <FormattedMessage
                  defaultMessage="Login to your account"
                  id="ocpui.components.Login.title"
                  values={Object {}}
                />
              }
            />
            <Formik
              enableReinitialize={false}
              isInitialValid={false}
              onSubmit={[Function]}
              render={[Function]}
              validateOnBlur={true}
              validateOnChange={true}
              validationSchema={
                ObjectSchema {
                  "_blacklist": Array [],
                  "_conditions": Array [],
                  "_defaultDefault": [Function],
                  "_deps": Array [],
                  "_excludedEdges": Array [],
                  "_exclusive": Object {},
                  "_mutate": false,
                  "_nodes": Array [
                    "password",
                    "username",
                  ],
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "object",
                  "_typeError": [Function],
                  "_whitelist": Array [],
                  "fields": Object {
                    "password": StringSchema {
                      "_blacklist": Array [],
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "required": false,
                      },
                      "_mutate": false,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "string",
                      "_typeError": [Function],
                      "_whitelist": Array [],
                      "tests": Array [
                        [Function],
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                    "username": StringSchema {
                      "_blacklist": Array [],
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "required": false,
                      },
                      "_mutate": false,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "string",
                      "_typeError": [Function],
                      "_whitelist": Array [],
                      "tests": Array [
                        [Function],
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                  },
                  "tests": Array [],
                  "transforms": Array [
                    [Function],
                  ],
                }
              }
            />
          </Styled(Card)>
        </Cell__Cell>,
        "columns": 3,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "center": true,
          "children": <Styled(Card)>
            <CardHeader
              title={
                <FormattedMessage
                  defaultMessage="Login to your account"
                  id="ocpui.components.Login.title"
                  values={Object {}}
                />
              }
            />
            <Formik
              enableReinitialize={false}
              isInitialValid={false}
              onSubmit={[Function]}
              render={[Function]}
              validateOnBlur={true}
              validateOnChange={true}
              validationSchema={
                ObjectSchema {
                  "_blacklist": Array [],
                  "_conditions": Array [],
                  "_defaultDefault": [Function],
                  "_deps": Array [],
                  "_excludedEdges": Array [],
                  "_exclusive": Object {},
                  "_mutate": false,
                  "_nodes": Array [
                    "password",
                    "username",
                  ],
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "object",
                  "_typeError": [Function],
                  "_whitelist": Array [],
                  "fields": Object {
                    "password": StringSchema {
                      "_blacklist": Array [],
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "required": false,
                      },
                      "_mutate": false,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "string",
                      "_typeError": [Function],
                      "_whitelist": Array [],
                      "tests": Array [
                        [Function],
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                    "username": StringSchema {
                      "_blacklist": Array [],
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "required": false,
                      },
                      "_mutate": false,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "string",
                      "_typeError": [Function],
                      "_whitelist": Array [],
                      "tests": Array [
                        [Function],
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                  },
                  "tests": Array [],
                  "transforms": Array [
                    [Function],
                  ],
                }
              }
            />
          </Styled(Card)>,
          "left": 2,
          "top": 3,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <CardHeader
                title={
                  <FormattedMessage
                    defaultMessage="Login to your account"
                    id="ocpui.components.Login.title"
                    values={Object {}}
                  />
                }
              />,
              <Formik
                enableReinitialize={false}
                isInitialValid={false}
                onSubmit={[Function]}
                render={[Function]}
                validateOnBlur={true}
                validateOnChange={true}
                validationSchema={
                  ObjectSchema {
                    "_blacklist": Array [],
                    "_conditions": Array [],
                    "_defaultDefault": [Function],
                    "_deps": Array [],
                    "_excludedEdges": Array [],
                    "_exclusive": Object {},
                    "_mutate": false,
                    "_nodes": Array [
                      "password",
                      "username",
                    ],
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "object",
                    "_typeError": [Function],
                    "_whitelist": Array [],
                    "fields": Object {
                      "password": StringSchema {
                        "_blacklist": Array [],
                        "_conditions": Array [],
                        "_deps": Array [],
                        "_exclusive": Object {
                          "required": false,
                        },
                        "_mutate": false,
                        "_options": Object {
                          "abortEarly": true,
                          "recursive": true,
                        },
                        "_type": "string",
                        "_typeError": [Function],
                        "_whitelist": Array [],
                        "tests": Array [
                          [Function],
                          [Function],
                        ],
                        "transforms": Array [
                          [Function],
                        ],
                      },
                      "username": StringSchema {
                        "_blacklist": Array [],
                        "_conditions": Array [],
                        "_deps": Array [],
                        "_exclusive": Object {
                          "required": false,
                        },
                        "_mutate": false,
                        "_options": Object {
                          "abortEarly": true,
                          "recursive": true,
                        },
                        "_type": "string",
                        "_typeError": [Function],
                        "_whitelist": Array [],
                        "tests": Array [
                          [Function],
                          [Function],
                        ],
                        "transforms": Array [
                          [Function],
                        ],
                      },
                    },
                    "tests": Array [],
                    "transforms": Array [
                      [Function],
                    ],
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "title": <FormattedMessage
                  defaultMessage="Login to your account"
                  id="ocpui.components.Login.title"
                  values={Object {}}
                />,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "enableReinitialize": false,
                "isInitialValid": false,
                "onSubmit": [Function],
                "render": [Function],
                "validateOnBlur": true,
                "validateOnChange": true,
                "validationSchema": ObjectSchema {
                  "_blacklist": Array [],
                  "_conditions": Array [],
                  "_defaultDefault": [Function],
                  "_deps": Array [],
                  "_excludedEdges": Array [],
                  "_exclusive": Object {},
                  "_mutate": false,
                  "_nodes": Array [
                    "password",
                    "username",
                  ],
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "object",
                  "_typeError": [Function],
                  "_whitelist": Array [],
                  "fields": Object {
                    "password": StringSchema {
                      "_blacklist": Array [],
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "required": false,
                      },
                      "_mutate": false,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "string",
                      "_typeError": [Function],
                      "_whitelist": Array [],
                      "tests": Array [
                        [Function],
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                    "username": StringSchema {
                      "_blacklist": Array [],
                      "_conditions": Array [],
                      "_deps": Array [],
                      "_exclusive": Object {
                        "required": false,
                      },
                      "_mutate": false,
                      "_options": Object {
                        "abortEarly": true,
                        "recursive": true,
                      },
                      "_type": "string",
                      "_typeError": [Function],
                      "_whitelist": Array [],
                      "tests": Array [
                        [Function],
                        [Function],
                      ],
                      "transforms": Array [
                        [Function],
                      ],
                    },
                  },
                  "tests": Array [],
                  "transforms": Array [
                    [Function],
                  ],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
